### PR TITLE
feat(core,query,prom): LP, ExecPlan, and _filodb_chunkmeta_all function to debug chunk metadata info

### PR DESCRIPTION
**Pull request by velvia**
**Pull Request checklist**

- [x] The commit(s) message(s) follows the contribution [guidelines](CONTRIBUTING.md) ?
- [x] Tests for the changes have been added (for bug fixes / features) ?
- [x] Docs have been added / updated (for bug fixes / features) ?

**New behavior :**

This PR creates a new special function, parseable through PromQL, `_filodb_chunkmeta_all` which returns `ChunkSetInfo` records for matching time series.  It returns all the fields from the ChunkSetInfo plus additional data (# bytes and the Reader class) for chunks in one particular column.  This data can be used to debug many issues including potential missing chunks or missing data, as well as to debug encoding and compression issues.

other related changes:
- One line minor fix for index startTime
- Fixed the standalone IngestionAndRecoverySpec; switched TestTimeseriesProducer to use the new Gateway pipeline

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -379,6 +383,18 @@ TODO: add details about FiloDB sharding, the shard-key vs partition key mechanis
...
+
+Current status:
+
+### FiloDB Extensions
+
+Some special functions exist to aid debugging and for other purposes:
+
+| function | description    |
+|----------|----------------|
+| `_filodb_chunkmeta_all` | (CLI Only) Returns chunk metadata fields for all chunks matching the time range and filter criteria - ID, # rows, start and end time, as well as the number of bytes and type of encoding used for a particular column.  |
```
Comment (vishramachandran) [inline README.md]: Can we add example usage from CLI ?

File: core/src/main/scala/filodb.core/store/ChunkSetInfo.scala [hunk 2]
```diff
@@ -274,4 +274,51 @@ class FilteredChunkInfoIterator(base: ChunkInfoIterator, filter: ChunkSetInfo =>
...
+/**
+ * A RowReader that returns info about each ChunkSetInfo set via setInfo with a schema like this:
+ * 0:  ID (Long)
+ * 1:  NumRows (Int)
+ * 2:  startTime (Long)
+ * 3:  endTime (Long)
+ * 4:  numBytes(Int) of chunk
+ * 5:  readerclass of chunk
+ */
+class ChunkInfoRowReader(column: Column) extends RowReader {
```
Comment (vishramachandran) [inline core/src/main/scala/filodb.core/store/ChunkSetInfo.scala]: Following additional information will be useful:
* encoding format for the chunk
* whether it is a write-buffer

Eventually, these can be added:
* Block information (odp or regular) 
* Lock status (how many readers) for debugging locking issues.

File: core/src/main/scala/filodb.core/memstore/TimeSeriesPartition.scala [hunk 3]
```diff
@@ -16,6 +16,8 @@ object TimeSeriesPartition extends StrictLogging {
   val nullChunks    = UnsafeUtils.ZeroPointer.asInstanceOf[AppenderArray]
   val nullInfo      = ChunkSetInfo(UnsafeUtils.ZeroPointer.asInstanceOf[BinaryRegion.NativePointer])
 
+  val _log = logger
```
[the remark this answers is not in the document]
Comment (velvia) [reply core/src/main/scala/filodb.core/memstore/TimeSeriesPartition.scala]: It was needed for debug logging in TSPartition.

File: README.md (hunk 1 above)
Comment (velvia) [reply README.md]: Sure.

File: core/src/main/scala/filodb.core/store/ChunkSetInfo.scala (hunk 2 above)
Comment (velvia) [reply core/src/main/scala/filodb.core/store/ChunkSetInfo.scala]: Yes definitely.   BTW the first two, encoding format for the chunk is what the "readerclass" is.  If the class is one of the uncompressed ones then it's a write buffer (there isn't really any other current way to tell, other than it being the first chunk).   

In future versions we can add a function which can derive the writebuffer-ness from the existing info...

agreed on block info

**Comment by velvia** [comment]
@vishramachandran comments addressed in latest commit.  Will merge sometime after PRB.

**Pull request merged**